### PR TITLE
Document validator results

### LINKED ISSUE #593: I want the developer documentation to document the rationale and structure of validation results
As an AaC developer, I want the developer documentation to document the rationale and structure of validation results, so that I can understand the design and intent of the system.

## AC:
- [ ] There is developer documentation that describes the general data returned during validations:
  - [ ] ValidatorFinding
  - [ ] FindingSeverity
  - [ ] FindingLocation
- [ ] There is a rationale provided for the data that is returned, indicating where and how it is used throughout AaC.
  - [ ] Include documentation on the use of valdiator findings within VSCode diagnostics functionality
  - [ ] Include documentation on the return of the CLI ValidationResult output

### PR DESCRIPTION
**Pull request by bunchw**
# Description

Includes a new section in aac validation documentation on validator results, which provides information on the return data of validators.

# Linked Items:

Closes #593 

# Checklist:

- [x] I updated project documentation to reflect my changes.
- [x] My changes generate no new warnings.
- [x] I updated new and existing unit tests to account for my changes.
- [x] I linked the associated item(s) to be closed.
- [x] I bumped the version.
- [x] I added the labels corresponding to my changes.

### CODE REVIEW

**Comment by lizzcondrey** [comment]
Version bump.

File: docs/source/project_documentation/dev_guide/aac_validation.md [hunk 1]
```diff
@@ -52,3 +52,11 @@ The internal logic of the function is up to the user, but the plugins generally
 3. If the definition under test doesn't meet the constraints of the validator, register and return an error message
 
 For more information on creating and utilizing validator plugins, please view the [Validation Plugins for Developers](validation_plugins).
+
+## Validation Results
+
+### Validator Findings
+A ValidatorFinding is returned when the validator wants to provide feedback of some kind.  A failed validation or error message is one type of feedback that can be provided, but it does not have to be limited to that.  As such, a FindingSeverity object is included to inform how severe the finding is.  A FindingLocation object will also be returned, which contains the source file and line number where the finding occured.
```
Comment (lizzcondrey) [inline docs/source/project_documentation/dev_guide/aac_validation.md:59]:
```suggestion
A `ValidatorFinding` is returned when the validator wants to provide feedback of some kind.  A failed validation or error message is one type of feedback that can be provided, but it does not have to be limited to that.  As such, a `FindingSeverity` object is included to inform how severe the finding is.  A `FindingLocation` object will also be returned, which contains the source file and line number where the finding occurred. 
```

**Review by lizzcondrey — COMMENTED** [review]
This is a solid summation that better explains the components of the result. And actually think it would be better placed within `validation_plugins` under the `Meanings of Validation Messages` section. Forgot which page that was captured on, thought it was here.

File: python/src/aac/__init__.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 import os
 
 
-__version__ = "0.3.17"
+__version__ = "0.3.19"
```
Comment (lizzcondrey) [inline python/src/aac/__init__.py:16]:
```suggestion
__version__ = "0.3.20"
```

have two PRs before this one that should be getting to go in.

**Review by lizzcondrey — CHANGES_REQUESTED** [review]

File: docs/source/project_documentation/dev_guide/aac_validation.md [hunk 3]
```diff
@@ -52,3 +52,11 @@ The internal logic of the function is up to the user, but the plugins generally
...
 
 For more information on creating and utilizing validator plugins, please view the [Validation Plugins for Developers](validation_plugins).
+
+## Validation Results
+
+### Validator Findings
+A ValidatorFinding is returned when the validator wants to provide feedback of some kind.  A failed validation or error message is one type of feedback that can be provided, but it does not have to be limited to that.  As such, a FindingSeverity object is included to inform how severe the finding is.  A FindingLocation object will also be returned, which contains the source file and line number where the finding occured. 
+
+### Validator Result
+After validation occurs, a ValidatorResult object will return the results of the validation, and compile the results into single message.  A ValidatorResult will contain a list of definitions that were validated, and a ValidatorFindings object wihich contains all ValidatorFinding objects returned by the validation.
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply docs/source/project_documentation/dev_guide/aac_validation.md:57]: Move this section over to the other page as well, sorry the highlight of the original comment wasn't very clear.

**Review by crazynewidea — APPROVED** [review]

**Review by lizzcondrey — APPROVED** [review]

**Pull request merged**